### PR TITLE
[hal-0.1] back-port fixes in dx12 and empty

### PR DESCRIPTION
**Pull request by kvark**
Backports a few essential changes in dx12 and empty backends.
PR checklist:
- [ ] `make` succeeds (on *nix)
- [x] `make reftests` succeeds
- [ ] tested examples with the following backends:
- [ ] `rustfmt` run on changed code

### CODE REVIEW

**Comment by kvark** [comment]
Note: I didn't end up carrying the dx12 vertex buffer binding fixes because Warden tests start panicing with it. cc @msiglreith

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 sudo: false
 language: rust
 cache:
-  cargo: true
   directories:
-   - $HOME/deps
+   - $HOME/.cargo
+before-cache:
+  - rm -rf $HOME/.cargo/registry
```
Comment (omni-viral) [inline .travis.yml]: Why do you need to do that?

Comment (kvark) [reply .travis.yml]: it was sorta sneaked it because everything was broken on Travis, see https://gitter.im/gfx-rs/gfx?at=5c4229f70a491251e31a3355

**Comment by msiglreith** [comment]
> Note: I didn't end up carrying the dx12 vertex buffer binding fixes because Warden tests start panicing with it. 

Could you post an issue with the errors? If feasible output from directx validation layers would be great also.

**Comment by kvark** [comment]
It will fail because of Appveyor. Added a commit disabling it. Fingers crossed ;)

Bors r=grovesNL

**Pull request merged**